src/coreclr/scripts/superpmi.py

```diff
@@ -2635,6 +2635,21 @@ def get_mch_files_for_replay(local_mch_paths, filters):
     return mch_files
 
 
+# Tries to determinate remote name for 'dotnet/runtime' upstream
+def get_upstream_name():
```

Comment:
To cover the case where upstream-ish remote does not exist (yet); maybe instead of locating the existing remote, just add remote with a known name (it's cheap as deltas across all remotes are downloaded idempotently; if-not-exist)? GitHub org/user name can be used as name of the remote, which is always unique and easiest to remember:

```sh
git remote add dotnet https://github.com/dotnet/runtime
git remote add egorbo https://github.com/EgorBo/runtime
git fetch --all
```

Reply:
This function to cover the case where upstream has other than 'origin' name, e.g. when you clone your fork from github `origin` points to your fork and you most likely has `dotnet` to point to `dotnet/runtime` upstream.

I personally always clone the upstream and then add a remote to my fork so on my machine `origin` targets `dotnet/runtime`.

Reply:
I understand. Considering the case, when someone have not added the upstream remote pointing to dotnet/runtime, we can add one in this function instead of returning `None`. My suggestion was that adding a remote that way (with predetermined name like `dotnet`) will also make finding the existing remote unnecessary. It is not expensive if the remote already existed with some other name, since git only downloads unique deltas across remotes; so it will be just two aliases pointing to same remote.

Reply:
@dotnet/jit-contrib any thoughts on that? `git remote add dotnet https://github.com/dotnet/runtime` feels a bit wrong to me because it modifies my environment where I already have `origin`

Reply:
Modifying the environment seems wrong, if the proposal is to do that.

I'm not sure how that will affect the way I set up branches. E.g., I have remotes:
```
origin  https://github.com/BruceForstall/runtime (fetch)             
origin  https://github.com/BruceForstall/runtime (push)              
upstream        https://github.com/dotnet/runtime (fetch)            
upstream        Don't push here. Fork and send PR.runtime (push)     
```
and currently `git merge-base xxx main` will my local main, which I keep up-to-date with my origin/main (Actually, maybe it should specify `origin/main` instead of `main`, to match jitrollingbuild.py?)

With the change, it will use `upstream/main` instead. But then I need to keep upstream fetched before running the command? (or at least fetch it when I create my branches?)